### PR TITLE
Preserve source location when using non-procs

### PR DESCRIPTION
**Pull request by zdrve**
Discussion:

The "Managed by kennel" line adds the tracking id, and then the project class's `file_location`.  `file_location` in turn is worked out by picking an essentially random instance method from the class (`instance_methods(false).first`) and using a processed form of its source_location.

This was a problem for the new proc-less form because the source location of the procs was inside the Kennel codebase (see `AS_PROCS`). Furthermore, the results were unstable, because of the randomness of the chosen instance method; if it picked one which was defined _with_ a proc, everything was fine, but if it picked one using the new proc-less syntax, then the source_location and therefore file_location and therefore "Managed by kennel" line, would be wrong.

### CODE REVIEW

**Review by grosser — APPROVED** [review]
🎉 much simpler :D

**Comment by grosser** [comment]
rubocop would like a work with you :)
... a test would also be nice so we see when we break it

File: lib/kennel/models/project.rb [hunk 1]
```diff
@@ -11,6 +11,8 @@ class Project < Base
       def self.file_location
         @file_location ||= begin
           method_in_file = instance_methods(false).first
+          return if method_in_file.nil?
```
Comment (grosser) [inline lib/kennel/models/project.rb:14]: that should never happen now right ?

**Comment by grosser** [comment]
1.114.1

**Comment by grosser** [comment]
... we should also check this new syntax performance wise just in case it's super slow

File: lib/kennel/models/project.rb (hunk 1 above)
Comment (zdrve) [reply lib/kennel/models/project.rb:14]: Super edge case, will never happen with normal data. See the test case.

**Comment by zdrve** [comment]
> in case it's super slow

Seems unlikely; the old syntax made procs and passed them to kennel; the new syntax passes values to kennel, which then makes procs. But yes it is of course an extra opcode or two just to do the detection.

**Comment by grosser** [comment]
FYI ran a small benchmark and there is no noticeable difference (loading 2k monitors with old or new syntax)
